azure/cargo-check.yml

```diff
@@ -33,17 +26,16 @@ jobs:
         echo '##vso[task.setvariable variable=is_locked]false';
       fi
     displayName: Are dependencies locked?
-  # we only check with --all here since we don't care about the actual
-  # compilation, just the dependencies + dev-dependencies resolution.
-  # we _do_ include all features as it may matter though
+  - script: cargo check --locked
```

Comment:
This should probably include `--all`.

Reply:
How come?

If its to ensure the lock file is correct, from my understanding the lock file is fully evaluated independent of `-all`.

Reply:
I'm not sure, but you could be right. My worry would be that without `--all`, dependencies of subcrates are not checked by `--locked` (since they are never entered in the first place). Do you have evidence to the contrary? Should be easy to test manually I suppose.

Reply:
Confirmed that `--all` is not needed
- Cloned `predicates-rs`
- `cargo check`
- `cargo check --locked`: success
- Edited predicates-tree/Cargo.toml to add a dep (this is not a dep of the root package)
- `cargo check --locked`: failed as expected.